operators/src/source/gdal_source/loading_info.rs

```
@@ -54,6 +53,134 @@ impl MetaData<GdalLoadingInfo, RasterResultDescriptor, RasterQueryRectangle>
+    ) -> Result<Self> {
+        let mut sorted_time_steps = time_steps;
+        sorted_time_steps.sort_by_key(TimeInterval::start);
+
+        let mut time_steps_no_gaps = sorted_time_steps.clone();
+
+        for (index, time) in sorted_time_steps.iter().enumerate() {
+            if index < sorted_time_steps.len() - 1
+                && time.end() < sorted_time_steps[index + 1].start()
+            {
+                time_steps_no_gaps.push(TimeInterval::new(
+                    time.end(),
+                    sorted_time_steps[index + 1].start(),
+                )?);
+            }
+        }
+
+        time_steps_no_gaps.sort_by_key(TimeInterval::start);
+
+        Ok(GdalMetadataFixedTimes {
+            time_steps: sorted_time_steps,
+            time_steps_no_gaps,
+            params,
+            result_descriptor,
+            time_placeholders,
+        })
+    }
+
+    fn create_temporal_slice(
+        &self,
+        step_interval: TimeInterval,
+        empty: bool,
+    ) -> Result<GdalLoadingInfoTemporalSlice> {
+        self.params
+            .replace_time_placeholders(&self.time_placeholders, step_interval)
+            .map(|loading_info_part_params| {
+                if empty {
+                    GdalLoadingInfoTemporalSlice {
+                        time: step_interval,
+                        params: None,
+                    }
+                } else {
+                    GdalLoadingInfoTemporalSlice {
+                        time: step_interval,
+                        params: Some(loading_info_part_params),
+                    }
+                }
+            })
+    }
+}
+
+#[async_trait]
+impl MetaData<GdalLoadingInfo, RasterResultDescriptor, RasterQueryRectangle>
+    for GdalMetadataFixedTimes
+{
+    async fn loading_info(&self, query: RasterQueryRectangle) -> Result<GdalLoadingInfo> {
+        let valid_duration = TimeInterval::new(
+            self.time_steps_no_gaps[0].start(),
```

this panics if the time_steps_no_gaps is empty. Please return an error instead.